docs/contributing/code_guide.rst

```diff
@@ -279,39 +279,138 @@ to edit the pull request by changing the corresponding branch on your
 PlasmaPy fork on GitHub.  After a pull request is merged into the
 code, you may delete the branch you created for that pull request.
 
-Commit Messages
----------------
-Good commit messages communicate context and intention to other
-developers and to our future selves.  They provide insight into why we
-chose a particular implementation, and help us avoid past mistakes.
 
-Suggestions on `how to write a git commit message
-<https://cbea.ms/git-commit>`_:
+Comments
```

Comment:
I wonder if establishing a "golden rule" would be useful here.  Maybe even put it into an admonition.  I personally get lost in a list of guide points, so having a golden rule to check against is what I find most useful.  The golden rule I tend to tell people goes something like:

> Your code should read like a book, that is the full meaning of your functionality should be attainable by reading the code.  A comment is only used when code itself can not communicate the full detailed meaning.

I'm sure there's a better way of phrasing this, but I think it covers all the points you say below.  I think pairing the "golden rule" with your points as examples would be overall more effective.

Thoughts?

Reply:
Good idea!  I'll add something like this in, likely after I get back from visiting family.